### PR TITLE
RM-7060: Fix exception being thrown when deselecting a tab preceding a…

### PR DESCRIPTION
**Pull request by iam-mholle**
…n invisible tab

### CODE REVIEW

File: Remotion/Web/UnitTests/Core/UI/Controls/WebTabCollectionTest.cs [hunk 1]
```diff
@@ -401,6 +401,34 @@ public void SelectInvisibleTab ()
...
+    _tabStrip.Tabs.Add(_tab1);
+    _tabStrip.Tabs.Add(_tab2);
+    _tab0.IsSelected = true;
+    _tab1.IsVisible = false;
+    Assert.That(_tabStrip.SelectedTab, Is.SameAs(_tab0));
+    _tab0.IsVisible = false;
+
+    Assert.That(_tabStrip.SelectedTab, Is.SameAs(_tab2));
+  }
+
```
Comment (MichaelKetting) [inline Remotion/Web/UnitTests/Core/UI/Controls/WebTabCollectionTest.cs:431]: Something from the tests to verify:
basically, the semantic is, to only select the first visible previous tab, if there is no visible next tab.

I think we need a test that shows that 
tab0.visbile=true
tab1.visibible=true
tab2.visible=false
tab3.visible=true
tab1.selected=true;
tab1.visible=false;
assert tab3.selected==true

**Review by MichaelKetting — CHANGES_REQUESTED** [review]

**Comment by iam-mholle** [comment]
The old behaviour should be restored, test cases should fit your requirements now.

File: Remotion/Web/Core/UI/Controls/WebTabCollection.cs [hunk 2]
```diff
@@ -148,23 +148,13 @@ protected void DeselectTab (WebTab tab, int index)
...
-          else
-            _tabStrip.SetSelectedTabInternal(null);
-        }
-        else
-        {
-          WebTab nextTab = (WebTab)InnerList[index + 1]!;
-          _tabStrip.SetSelectedTabInternal(nextTab);
-        }
+        var webTabs = InnerList.Cast<WebTab>().ToList();
+        var nextTab = webTabs.Skip(index + 1).FirstOrDefault(IsSelectable) ?? webTabs.Take(index).LastOrDefault(IsSelectable);
```
Comment (MichaelKetting) [inline Remotion/Web/Core/UI/Controls/WebTabCollection.cs:152]: readability and allocation prevention (the ToList isn't needed, we can just to a cast twice):
```suggestion
        var tabsBeforeSelectedTab = InnerList.Cast<WebTab>().Take(index);
        var tabsAfterSelectedTab = InnerList.Cast<WebTab>().Skip(index + 1);
        var nextTab = tabsAfterSelectedTab.FirstOrDefault(IsSelectable) ?? tabsBeforeSelectedTab .LastOrDefault(IsSelectable);
```

If we didn't incur the allocation cost for the array, we could also do C# Range syntax.

**Review by MichaelKetting — APPROVED** [review]
Readability optimization

**Pull request merged**